File: Swift-C++-Testsuite/test/concurrency/Swift/BasicAsync.swift
@main
struct MyMain {

    static func main() async {
        let photos = await listPhotos(inGallery: "home")
    }

    static func listPhotos(inGallery name: String) async -> [String] {
        let result = ["MyPhotoNames"]  // ... some asynchronous networking code ...
        do {
            try await Task.sleep(until: .now + .seconds(2), clock: .continuous)
        } catch {
            print("Sleep failed")
        }
        return result
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/BasicAsync.cpp

int main() {
    auto photos = listPhotos("home");
    return 0;
}

std::vector<std::string> listPhotos(const std::string &name) {
  std::vector<std::string> result = {"MyPhotoNames"};  // ... some asynchronous networking code ...
  std::this_thread::sleep_for(std::chrono::seconds(2));
  return result;
}